packages/xforms-engine/test/parse/xpath/semantic-analysis.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expressionParser } from '@getodk/xpath/expressionParser.js';
+import { describe, expect, it } from 'vitest';
+import {
+	findLocationPathSubExpressionNodes,
+	isTranslationExpression,
+} from '../../../src/parse/xpath/semantic-analysis.ts';
+
+describe('Semantic analysis', () => {
+	describe('FunctionCall', () => {
+		describe('translation', () => {
+			interface IsTranslationExpressionCase {
+				readonly description: string;
+				readonly expression: string;
+				readonly expected: boolean;
+			}
+
+			it.each<IsTranslationExpressionCase>([
```

Comment:
I was expecting following case to pass:

```
{
  description: 'argument is a string',
  expression: 'jr:itext(string(1))',
  expected: true,
},
```

Reply:
My first thought looking at this comment was "well yeah, and it should also accept anything else returning or producing a string". And then it occurred to me that... that's everything. Even the test cases where we currently reject some argument types (e.g. Number), because the `xpath` evaluator will cast them to a string. So the more I think about this, the more I think we shouldn't bother checking anything about the Argument's expression at all, except that it's there.

Reply:
true, I was not sure if we expect non-core functions to be doing explicit casting of the arguments.